kie-server-parent/kie-server-services/kie-server-services-task-assigning/kie-server-services-task-assigning-planning/src/main/java/org/kie/server/services/taskassigning/planning/SolverHandler.java

```diff
@@ -215,13 +295,32 @@ private void onSolutionProcessed(SolutionProcessor.Result result) {
 
     /**
      * Invoked every time the SolutionSynchronizer gets updated information from the jBPM runtime and there are changes
-     * to apply.
+     * to apply, or when the configurable amount of time with no changes has elapsed.
      * @param result Contains the list of changes to apply.
      */
-    private void onUpdateSolution(SolutionSynchronizer.Result result) {
+    private void onSolutionSynchronized(SolutionSynchronizer.Result result) {
         lock.lock();
         try {
-            addProblemFactChanges(result.getChanges());
+            if (result.hasChanges()) {
+                addProblemFactChanges(result.getChanges());
+            } else {
+                LOGGER.debug("Processing synchronization unchanged period timeout. Checking if there is a" +
+                                     " lastBestSolution with an improved score to send");
+                TaskAssigningSolution bestSolution = lastBestSolution.get();
+                secondLevelSolutionSent.set(true);
+                if (currentSolution.get() != bestSolution && !currentSolution.get().getScore().equals(bestSolution.getScore())) {
```

Comment:
Do we need the compound condition? Maybe we should send the new best solution if its score is better than the score of the current solution.